File: bcube-frontend/src/app/models/studio.ts
import { User } from "./user";

export interface studio {
    id: number;
    name: string;
    description: string;
    street: string;
    plz: number;
    isActive: boolean;
    city: string;              
    country: string;           
    latitude?: number | null;
    longitude?: number | null;
    image: number[];
    imageBase64: string;
    users: User[];
}

File: bcube-ios/bcube-ios/Models/Studio.swift
import Foundation
import CoreLocation

struct Studio: Codable, Identifiable, Hashable {
    let id: Int
    let name: String
    let description: String
    let street: String
    let plz: Int
    let isActive: Bool?
    let city: String
    let country: String
    let latitude: Double?
    let longitude: Double?
    let image: [Int]?
    let imageBase64: String

    var fullAddress: String {
        "\(street), \(plz) \(city)"
    }

    var coordinate: CLLocationCoordinate2D? {
        guard let latitude, let longitude else { return nil }
        return CLLocationCoordinate2D(latitude: latitude, longitude: longitude)
    }
}

struct StudioFilter: Codable, Identifiable, Hashable {
    let id: Int
    let name: String
}
